query/query.go

```diff
@@ -175,6 +176,38 @@ func (q *Repo) String() string {
 	return fmt.Sprintf("repo:%s", q.Pattern)
 }
 
+// RepoRegexp is a Sourcegraph addition which searches documents where the
+// repository name matches Regexp.
+type RepoRegexp struct {
+	Regexp *regexp.Regexp
+}
+
+func (q *RepoRegexp) String() string {
+	return fmt.Sprintf("reporegex:%q", q.Regexp.String())
+}
+
+// GobEncode implements gob.Encoder.
+func (q *RepoRegexp) GobEncode() ([]byte, error) {
+	// gob can't encode syntax.Regexp
+	var buf bytes.Buffer
```

Comment:
nit: you could save the extra allocations and just do: `return []byte(q.Regexp.String()), nil`

Reply:
I want to encode the version just in case we change it in the future. However, I suppose we could just introduce a new type? WDYT?

Reply:
`return []byte("1" + q.Regexp.String()), nil` YOLO